### PR TITLE
optional disable nspv

### PR DESCRIPTION
**Pull request by ca333**

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -430,6 +430,7 @@ std::string HelpMessage(HelpMessageMode mode)
     strUsage += HelpMessageOpt("-onlynet=<net>", _("Only connect to nodes in network <net> (ipv4, ipv6 or onion)"));
     strUsage += HelpMessageOpt("-permitbaremultisig", strprintf(_("Relay non-P2SH multisig (default: %u)"), 1));
     strUsage += HelpMessageOpt("-peerbloomfilters", strprintf(_("Support filtering of blocks and transaction with Bloom filters (default: %u)"), 1));
+    strUsage += HelpMessageOpt("-nspv_msg", _("Enable NSPV messages processing"));
```
Comment (DeckerSU) [inline src/init.cpp:433]:
```suggestion
    strUsage += HelpMessageOpt("-nspv_msg", strprintf(_("Enable NSPV messages processing (default: %u)"), DEFAULT_NSPV_PROCESSING));
```
In `main.h` we should also include:
```
/** Default NSPV support enabled */
static const bool DEFAULT_NSPV_PROCESSING = false;
```

File: src/main.cpp [hunk 2]
```diff
@@ -7530,25 +7530,19 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
...
-    }
-    else if (strCommand == "nSPV")
-    {
-        if ( KOMODO_NSPV_SUPERLITE )
-        {
-            std::vector<uint8_t> payload;
-            vRecv >> payload;
-            komodo_nSPVresp(pfrom,payload);
+    // temporary optional nspv message processing
+    else if (GetBoolArg("-nspv_msg", false) &&
```
Comment (DeckerSU) [inline src/main.cpp:7534]:
```suggestion
    else if (GetBoolArg("-nspv_msg", DEFAULT_NSPV_PROCESSING) &&
```

File: src/net.cpp [hunk 3]
```diff
@@ -83,7 +83,7 @@ extern char ASSETCHAINS_SYMBOL[65];
 
 bool fDiscover = true;
 bool fListen = true;
-uint64_t nLocalServices = NODE_NETWORK | NODE_NSPV;
+uint64_t nLocalServices = GetBoolArg("-nspv_msg", false) ? NODE_NETWORK | NODE_NSPV : NODE_NETWORK;
```
Comment (DeckerSU) [inline src/net.cpp:86]:
```suggestion
uint64_t nLocalServices = GetBoolArg("-nspv_msg", DEFAULT_NSPV_PROCESSING) ? NODE_NETWORK | NODE_NSPV : NODE_NETWORK;
```

File: src/protocol.cpp [hunk 4]
```diff
@@ -102,7 +102,7 @@ CAddress::CAddress(CService ipIn, uint64_t nServicesIn) : CService(ipIn)
 
 void CAddress::Init()
 {
-    nServices = NODE_NETWORK | NODE_NSPV;
+    nServices = GetBoolArg("-nspv_msg", false) ? NODE_NETWORK | NODE_NSPV : NODE_NETWORK;
```
Comment (DeckerSU) [inline src/protocol.cpp:105]:
```suggestion
    nServices = GetBoolArg("-nspv_msg", DEFAULT_NSPV_PROCESSING) ? NODE_NETWORK | NODE_NSPV : NODE_NETWORK;
```

**Comment by dimxy** [comment]
Could we also use a global bool flag (like fNSPVSupported) instead of calling GetBoolArg("-nspv_msg", false)?
just to make code more efficient and be consistent with other cases where similar flags are used (fAddressIndex, fUspentIndex etc)

**Comment by DeckerSU** [comment]
> Could we also use a global bool flag (like fNSPVSupported) instead of calling GetBoolArg("-nspv_msg", false)?
> just to make code more efficient and be consistent with other cases where similar flags are used (fAddressIndex, fUspentIndex etc)

`fNSPVSupported` from one side is good idea, but actually we can check support of nSPV on local node with `if (nLocalServices & NODE_NSPV)` condition. If `nLocalServices` have `NODE_NSPV` flag - it's supported nspv. Same way as we already did it for `NODE_BLOOM` services flag, like:
```
else if (!(nLocalServices & NODE_BLOOM) &&
              (strCommand == "filterload" ||
               strCommand == "filteradd"))
{
...    
}
```

**Pull request merged**